### PR TITLE
Game Level Scriptable Object

### PR DESCRIPTION
**Pull request by Shadowmage77**
- Refactored game level prefab into a scriptable level object
- Truescene change is just the reference to a game level
- Removed prefab 'Levels' folder and added EditorObject 'Levels' folder
- Moved GameLevel script to Code/EditorObject folder

### CODE REVIEW

File: Assets/Code/Scripts/EditorObject/GameLevel.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using EditorObject;
+using UnityEngine;
+using UnityEngine.UI;
+
```
Comment (toryjleo) [inline Assets/Code/Scripts/EditorObject/GameLevel.cs:6]: Please put in `namespace EditorObject`

File: Assets/Scenes/Immutable/TrueScene.unity [hunk 2]
```diff
@@ -670,7 +670,7 @@ MonoBehaviour:
   m_EditorClassIdentifier: 
   jukebox: {fileID: 1714521336}
   worldGenerator: {fileID: 2121570724}
-  currentLevel: {fileID: 6967268396237467457, guid: b460f26878a4c2b4b99aa92d8cce33ab, type: 3}
```
Comment (toryjleo) [inline Assets/Scenes/Immutable/TrueScene.unity:673]: What was updated here?

**Review by toryjleo — CHANGES_REQUESTED** [review]

**Comment by toryjleo** [comment]
Pretty good work!

File: Assets/Scenes/Immutable/TrueScene.unity (hunk 2 above)
Comment (Shadowmage77) [reply Assets/Scenes/Immutable/TrueScene.unity:673]: LevelManager reference to a different level object that was a prefab.

**Review by toryjleo — APPROVED** [review]

**Comment by toryjleo** [comment]
LGTM :shipit:

**Pull request merged**